### DISCUSSION TITLE
Centroidal tessellation and termination criterion

### DISCUSSION THREAD

**Discussion by momohu** [neperfepx/neper#149] [Q&A] [answered]
I saw in an article that the distribution state can be adjusted by controlling λ, but I didn't find the corresponding function in the manual. What command is needed to realize this
neper -T -n 15 -domain "sphere(0.016,12):scale(1,1,1)"    -o sphere -format geo
https://link.springer.com/article/10.1007/s40571-016-0129-0
![image](https://user-images.githubusercontent.com/59075419/111796799-7ffb1280-8903-11eb-9b21-5112ea7bd5ce.png)
![image](https://user-images.githubusercontent.com/59075419/111796846-89847a80-8903-11eb-92a2-646ea413623c.png)

**Comment by rquey** [comment, marked as answer]
This is a centroidal tessellation, which you can get using `-morpho centroidal` or, equivalently, `-morpho "centroid:seed"`.

The expression of the termination criterion is a L1 norm. It is curious that it is not normalized by the number of cells, but I'm quite sure that the authors did normalize.

Neper's default is a L2 norm (normalized), but you can change it to the L1 norm (normalized) using `-morphooptiobj centroid:L1`.

The lambda value can be set using `-morpho val=...`.

Put this in a spherical domain and you get something like your figures:

```
$ neper -T -domain "sphere(1)" -n 250 -morpho centroidal -morphooptiobj centroid:L1 -morphooptistop "val=0.5"
$ neper -V n250-id1.tess -datacellcol id -print lambda-0p500

$ neper -T -domain "sphere(1)" -n 250 -morpho centroidal -morphooptiobj centroid:L1 -morphooptistop "val=0.005"
$ neper -V n250-id1.tess -datacellcol id -print lambda-0p005
```
![image](https://user-images.githubusercontent.com/2659376/111815243-bb81e680-88db-11eb-9178-edbf0a5937e3.png)
